File: chavticks.ui/src/app/responses/flight/FlightEventResponse.ts
import {AirportSummaryResponse} from "../airport/AirportSummaryResponse";

export class FlightEventResponse {
  airport: AirportSummaryResponse;
  scheduledTimeLocal: Date | null;
  runwayTimeLocal: Date | null;
  actualTimeLocal: Date | null;
  scheduledTimeUtc: Date | null;
  actualTimeUtc: Date | null;
  runwayTimeUtc: Date | null;
  terminal: string | null;
  checkInDesk: string | null;
  gate: string | null;
  baggageBelt: string | null;
  runway: string | null;
  quality: string[];
}

File: ChAvTicks.Application/Responses/Flight/Common/FlightEventResponse.cs
using ChAvTicks.Application.Responses.Airport.Common;

namespace ChAvTicks.Application.Responses.Flight.Common
{
	public record FlightEventResponse(
		DateTime? ScheduledTimeLocal,
		DateTime? ActualTimeLocal,
		DateTime? RunwayTimeLocal,
		DateTime? ScheduledTimeUtc,
		DateTime? ActualTimeUtc,
		DateTime? RunwayTimeUtc,
		string? Terminal,
		string? CheckInDesk,
		string? Gate,
		string? BaggageBelt,
		string? Runway,
		string[]? Quality)
	{
		public AirportSummaryResponse? Airport { get; set; }
	}
}
